### PR TITLE
Master coverity

### PR DESCRIPTION
**Pull request by Marek77**
Hello Tobi,

we are using rrdtool at work and as we have Coverity for static code analysis we want to contribute back :-)

Please don't hesitate to contact me if you have doubts with some of the commits, the log messages could be more verbose especially for the less-than obvious fixes.

regards
Marek

### CODE REVIEW

File: src/rrd_cgi.c [hunk 1]
```diff
@@ -418,7 +418,12 @@ static int readfile(
         totalcnt = (ftell(input) + 1) / sizeof(char) - offset;
         if (totalcnt < MEMBLK)
             totalcnt = MEMBLK;  /* sanitize */
-        fseek(input, offset * sizeof(char), SEEK_SET);
+        if (fseek(input, offset * sizeof(char), SEEK_SET) == -1)
```
Comment (stevenh) [inline src/rrd_cgi.c]: Slight style issue here.

File: src/rrd_cgi.c [hunk 2]
```diff
@@ -1379,8 +1406,10 @@ s_var   **rrdcgiReadVariables(
             length = atoi(ip);
             if ((line = (char *) malloc(length + 2)) == NULL)
                 return NULL;
-            if (fgets(line, length + 1, stdin) == NULL)
+            if (fgets(line, length + 1, stdin) == NULL) {
+            	free(line);
```
Comment (stevenh) [inline src/rrd_cgi.c]: Indentation (could be github visualisation of it though)

File: src/rrd_dump.c [hunk 3]
```diff
@@ -569,6 +569,7 @@ int rrd_dump(
                           "[--no-header|-n]\n"
                           "[--daemon|-d address]\n"
                           "file.rrd [file.xml]", options.argv[0]);
+            if (opt_daemon) free(opt_daemon);
```
Comment (stevenh) [inline src/rrd_dump.c]: Personally no a fan of this style of shorthand but not sure what the project rules are about it. That and it should be a NULL test.
```c
if (opt_daemon != NULL)
   free(opt_daemon);
```

Comment (oetiker) [inline src/rrd_dump.c]: *personally* I would even make the curly brackets ... :) as you can see the code is rather 'old' by now ...

**Comment by oetiker** [comment]
cool! thank you very much ... while on the subject ... isn't there some way to integrate coverty on travis and they hand out free licenses to oss projects ... do you have any experience with that ?

File: src/rrd_cgi.c (hunk 1 above)
Comment (Marek77) [inline src/rrd_cgi.c]: Please expand?

File: src/rrd_dump.c (hunk 3 above)
Comment (Marek77) [inline src/rrd_dump.c]: I have seen all kinds of code style in the source ;-) but I'm happy to adapt. Let me know..

**Comment by Marek77** [comment]
Sorry no idea how Github + Travis + Coverity works...

File: src/rrd_dump.c (hunk 3 above)
Comment (oetiker) [inline src/rrd_dump.c]: hihi ... yes ... so please ... braces everywhere ...

File: src/rrd_cgi.c (hunk 2 above)
Comment (Marek77) [inline src/rrd_cgi.c]: Yeah my editor removes all whitespace so I had to hand-edit some commits not to add a lot of whitespace diff. There should be 4 spaces there.

**Comment by oetiker** [comment]
is there a coverty configuration file ? or something like that ?

File: src/rrd_xport.c [hunk 4]
```diff
@@ -444,15 +444,17 @@ int rrd_graph_xport(image_desc_t *im) {
...
   }
+
+  /* free legend */
+  for (unsigned long j = 0; j < col_cnt; j++) {
+    free(legend_v[j]);
+  }
+  free(legend_v);
+  /* free data */
+  free(data);
+
```
Comment (oetiker) [inline src/rrd_xport.c]: are you sure that it is ok todo this in any event (regardles of r?)

Comment (Marek77) [inline src/rrd_xport.c]: Actually if (r != 0), free(data) is done and then return; however "data" is allocated by rrd_xport_fn() (line 407), (r==0) (switch(im->imgformat) case default)  but not free'd without this patch.

**Comment by Marek77** [comment]
I don't know about the configuration files; we have a coverity installation and "just use it" :-) (but I need to have a product / rrdtool,... / configured in coverity) but it's the admin team who manages it. Then I send my "make install" via cov-build wrapper.

**Comment by Marek77** [comment]
I have added curly braces ;-) and fixed the indent as per stevenh's comments.

**Pull request merged**